app/models/withdraw.rb

```diff
@@ -41,7 +41,7 @@ class Withdraw < ApplicationRecord
             numericality: { greater_than_or_equal_to: ->(withdraw) { withdraw.currency.min_withdraw_amount }}
 
   validate do
-    errors.add(:beneficiary, 'not active') if beneficiary.present? && !beneficiary.active?
+    errors.add(:beneficiary, 'not active') if beneficiary.present? && !beneficiary.active? && aasm.to_state != :failed
```

Comment:
```suggestion
    errors.add(:beneficiary, 'not active') if beneficiary.present? && !beneficiary.active? && failed?
```

I would even allow non-active beneficiary for all COMPLETED_STATES